osquery/tables/system/linux/portage.cpp

```diff
@@ -0,0 +1,326 @@
+    }
+    packageStr.erase(0, 1);
+  }
+  auto divPos = packageStr.find('/');
+  auto verPos = packageStr.find_last_of('-');
+  if (verPos != std::string::npos && verPos > divPos &&
+      verPos < packageStr.length() - 1 && isdigit(packageStr[verPos + 1])) {
+    package.second += packageStr.substr(verPos + 1);
+    packageStr.erase(verPos, packageStr.length() - verPos);
+  } else {
+    verPos = packageStr.find_last_of('-', verPos - 1);
+    if (verPos != std::string::npos && verPos > divPos &&
+        verPos < packageStr.length() - 1 && isdigit(packageStr[verPos + 1])) {
+      package.second += packageStr.substr(verPos + 1);
+      packageStr.erase(verPos, packageStr.length() - verPos);
+    }
+  }
+  package.first = packageStr;
+  return package;
+}
+
+/* Functions to fetch content for database tables */
+QueryData parsePortagePackages(const std::vector<std::string>& pkgPaths,
+                               const std::string& worldContent) {
+  QueryData results;
+
+  // store world in a map so we can use find to see if
+  // a package is in the world file
+  std::unordered_map<std::string, bool> world;
+  if (!worldContent.empty()) {
+    for (const auto& i : osquery::split(worldContent, "\n")) {
+      auto line = split(i);
+      if (line.size() == 0 || boost::starts_with(line[0], "#")) {
+        continue;
+      }
+      auto package = portageSplitPackageVersion(line[0]);
+      world.emplace(std::make_pair(package.first, true));
+    }
+  }
+
+  for (auto& directory : pkgPaths) {
+    Row r;
+    std::string pkg = directory;
+    pkg.erase(0, 12);
+
+    std::pair<std::string, std::string> package =
+        portageSplitPackageVersion(pkg);
+    r["package"] = package.first;
+    r["version"] = package.second;
+    r["world"] = world.end() == world.find(package.first) ? "0" : "1";
+
+    std::string slotCont;
+    if (readFile(directory + "/SLOT", slotCont).ok()) {
+      boost::trim(slotCont);
+      r["slot"] = slotCont;
+    }
+
+    std::string sizeCont;
+    if (readFile(directory + "/SIZE", sizeCont).ok()) {
+      boost::trim(sizeCont);
+      r["size"] = sizeCont;
+    }
+
+    std::string eapiCont;
+    if (readFile(directory + "/EAPI", eapiCont).ok()) {
+      boost::trim(eapiCont);
+      r["eapi"] = eapiCont;
+    }
+
+    std::string buildCont;
+    if (readFile(directory + "/BUILD_TIME", buildCont).ok()) {
+      boost::trim(buildCont);
+      r["build_time"] = buildCont;
+    }
+
+    std::string repoCont;
+    if (readFile(directory + "/repository", repoCont).ok()) {
+      boost::trim(repoCont);
+      r["repository"] = repoCont;
+    }
+
+    results.push_back(r);
+  }
+
+  return results;
+}
+
+QueryData parsePortageKeywordSummaryContent(const std::string& keywords,
+                                            const std::string& masked,
+                                            const std::string& unmasked) {
+  QueryData results;
+  std::unordered_map<std::string, PortagePackage*> tempStorage;
```

Comment:
@theopolis 
This was to avoid the copy so we could use boost::non_copyable for the PortagePackage.
If you have a suggestion on how to do this without copy and raw pointers, I would happy to change it again.